storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/BatchWriter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server.kvstore.dataaccess;
+
+import java.util.function.Consumer;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.teku.storage.server.kvstore.ColumnEntry;
+import tech.pegasys.teku.storage.server.kvstore.KvStoreAccessor;
+import tech.pegasys.teku.storage.server.kvstore.schema.KvStoreColumn;
+
+abstract class BatchWriter implements AutoCloseable {
+  private static final int OUTPUT_PER_TRANSACTION_COUNT = 15;
+  final KvStoreAccessor db;
+
+  private final Consumer<String> logger;
+  private final long targetBatchSize;
+
+  private long transactionCounter = 0;
+  private long entryCounter = 0;
+  private long bytes = 0;
+
+  /**
+   * @param targetBatchSize target size (MB) for a batch
+   * @param db database accessor object
+   */
+  BatchWriter(final int targetBatchSize, final Consumer<String> logger, final KvStoreAccessor db) {
+    // target batch size comes in MB, can store in Bytes to make life simpler for comparison
+    this.targetBatchSize = targetBatchSize * 1_000_000L;
+    this.logger = logger;
+    this.db = db;
+  }
+
+  abstract void startTransaction();
+
+  abstract KvStoreAccessor.KvStoreTransaction getTransaction();
+
+  abstract void commit();
```

Comment:
Since the subclasses are being given a `KvStoreAccessor` I don't think we need the subclasses at all. We can create a transaction with `KvStoreAccessor.startTransaction` and the accessor has the `putRaw` and `commit` methods we need.  The subclasses would only be necessary if we needed to use the `HotUpdater` wrapper classes, but we're down below that level so can just work work directly with the KvStore classes.

Reply:
that's a really good pickup! I'll leave the functions (startTransaction, getTransaction, commit) in case we ever need them, but remove the subclasses, and this will simplify life a lot!